packages/tradershub/src/providers/SignupWizardProvider/ValuesReducer.ts

```diff
@@ -17,6 +17,11 @@ export function valuesReducer(state: TState, action: TActions) {
                 firstName: payload.firstName,
                 lastName: payload.lastName,
             };
+        case ACTION_TYPES.SET_ADDRESS:
```

Comment:
@thisyahlen-deriv update ACTION_TYPES in SignupWizardContext and add SET_ADDRESS

Reply:
@thisyahlen-deriv also I have suggestion to make it easier to read, please, modify this if you think it's a good idea. I've asked Shahzaib about this but he said he will do it later when we have more actions. Maybe would be better to write like this:

```
    switch (type) {
        case ACTION_TYPES.SET_CURRENCY:
        case ACTION_TYPES.SET_PERSONAL_DETAILS:
        case ACTION_TYPES.SET_ADDRESS:
            return {
                            ...state,
                            ...payload,
                        };
         default:
             return state;
        }
```

Reply:
We will refactor this on screen based action. For now we just doing the screens

Reply:
It's not about screens, it's about grouping actions which has the same state... But it was just a suggestion, you can ignore it, ofc

Reply:
@sergei-deriv it is for now. The actions will be updated with their specific acceptable payload type and it'll be more explicit about the payloads

Reply:
Ah i see @sergei-deriv but then this could lead to bugs if, for example, the payload for SET_CURRENCY accidentally contains a firstName property. In the original code, this would be ignored, but in the proposed change, it would overwrite the firstName in the state i think

Reply:
@thisyahlen-deriv I understood what you mean and agree with you here. But in this case why our payload type allows us to pass wrong payload with specific action? :) I don't suggest to modify it and payload type for every action type, but it's something for what TS was build )))))) Anyway, please, resolve this conversation

Reply:
I agree with you @sergei-deriv. We will change this in the next PR.